tests/CMakeLists.txt

```diff
@@ -75,40 +75,48 @@ foreach (dbcsr_test ${DBCSR_TESTS})
 endforeach ()
 
 if (USE_CUDA)
+
+  # All libcusmm tests
+  set(LIBCUSMM_TESTS_BUILD
+    libcusmm_unittest_multiply
+    libcusmm_unittest_transpose
+    libcusmm_timer_multiply
+    )
+
+  # Tests that need no additional arguments to be run
   set(LIBCUSMM_SIMPLE_TESTS
     libcusmm_unittest_multiply
     libcusmm_unittest_transpose
     )
 
-  set(LIBCUSMM_TESTS_BUILD
-    ${LIBCUSMM_SIMPLE_TESTS}
+  # Tests whose source needs to be generated from a template
+  set(LIBCUSMM_TESTS_TO_GENERATE
+    libcusmm_unittest_multiply
     libcusmm_timer_multiply
     )
 
-  add_custom_command(
-    OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/libcusmm_unittest_multiply.cu
-    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/generate_libcusmm_unittest_multiply.py --base_folder ${CMAKE_CURRENT_SOURCE_DIR}/.. --gpu_version=${WITH_GPU}
-    DEPENDS libcusmm_unittest_multiply.template generate_libcusmm_unittest_multiply.py
-    COMMENT "Generate tests/libcusmm_unittest_multiply.cu"
-  )
-
-  add_custom_command(
-    OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/libcusmm_timer_multiply.cu
-    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/generate_libcusmm_timer_multiply.py --base_folder ${CMAKE_CURRENT_SOURCE_DIR}/.. --gpu_version=${WITH_GPU}
-    DEPENDS libcusmm_timer_multiply.template generate_libcusmm_timer_multiply.py
-    COMMENT "Generate tests/libcusmm_timer_multiply.cu"
-  )
-
-  foreach (libcusmm_test ${LIBCUSMM_TESTS_BUILD})
-    add_executable(${libcusmm_test} ${CMAKE_CURRENT_SOURCE_DIR}/${libcusmm_test}.cu)
+  # Add custom commands for the test files that need to be generated from a template
+  foreach (libcusmm_test ${LIBCUSMM_TESTS_TO_GENERATE})
+    add_custom_command(
+      OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${libcusmm_test}.cu
+      COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/generate_${libcusmm_test}.py --base_dir ${CMAKE_CURRENT_SOURCE_DIR}/.. --out_dir ${CMAKE_CURRENT_BINARY_DIR} --gpu_version=${WITH_GPU}
```

Comment:
According to the [cmake documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_BINARY_DIR.html) `$CMAKE_CURRENT_BINARY_DIR` is a full path. But below you're expecting `--out-dir` to be relative to `--base_dir`.

It seems this works because Python will return only the second argument of `os.path.join(a, b)` if `b` is an absolute path.

Reply:
Solved by calculating the relative path using [`file`](https://cmake.org/cmake/help/v3.4/command/file.html)

Reply:
ah, okay, sorry, didn't see that 🙈

Reply:
No no, you were right. I just fixed it in my last commit. 
Thanks for the review!